### ISSUE TITLE
Does not handle redis connections with auto-decoding on

### ISSUE THREAD

**Issue by mlasevich** [brainix/pottery#456]
**Describe the bug**
Redis client  has option to auto-decode responses (`decode_responses`) - when it is set to true, it seems to conflict with
pottery

**To Reproduce**
```
>>> from redis import StrictRedis
>>> from pottery import RedisDict
>>> conn = StrictRedis(host='redis', charset="utf-8", decode_responses=True)
>>> tel =  RedisDict({'jack': 4098, 'sape': 4139}, redis=conn, key='tel2')
>>> tel
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/pottery/dict.py", line 116, in __repr__
    dict_ = {self._decode(key): self._decode(value) for key, value in items}
  File "/usr/local/lib/python3.9/site-packages/pottery/dict.py", line 116, in <dictcomp>
    dict_ = {self._decode(key): self._decode(value) for key, value in items}
  File "/usr/local/lib/python3.9/site-packages/pottery/base.py", line 154, in _decode
    decoded: JSONTypes = json.loads(value.decode('utf-8'))
AttributeError: 'str' object has no attribute 'decode'
>>> 
```
**Expected behavior**
I expect it figure out when to decode

**Environment (please complete the following information):**
 - OS: MacOS
 - Python version : 3.9.7

**Comment by brainix** [comment]
Hi, @mlasevich.

Thanks for the bug report. I've fixed the bug in this PR: https://github.com/brainix/pottery/pull/458

This is what happens with Pottery 1.3.5:

```python
>>> from redis import Redis
>>> from pottery import RedisDict
>>> redis = Redis(decode_responses=True)
>>> tel = RedisDict({'jack': 4098, 'sape': 4139}, redis=redis, key='tel')
>>> tel
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rajiv.shah/Documents/Code/pottery/pottery/dict.py", line 116, in __repr__
    dict_ = {self._decode(key): self._decode(value) for key, value in items}
  File "/Users/rajiv.shah/Documents/Code/pottery/pottery/dict.py", line 116, in <dictcomp>
    dict_ = {self._decode(key): self._decode(value) for key, value in items}
  File "/Users/rajiv.shah/Documents/Code/pottery/pottery/base.py", line 154, in _decode
    decoded: JSONTypes = json.loads(value.decode('utf-8'))
AttributeError: 'str' object has no attribute 'decode'
```

And this is what happens with Pottery 1.3.6 (unreleased):

```python
>>> from redis import Redis
>>> from pottery import RedisDict
>>> redis = Redis(decode_responses=True)
>>> tel = RedisDict({'jack': 4098, 'sape': 4139}, redis=redis, key='tel')
>>> tel
RedisDict{'jack': 4098, 'sape': 4139}
```

How does the PR look? I'll release Pottery 1.3.6 soon.

Thanks again,
Raj

**Comment by mlasevich** [comment]
Looks nice. Thanks for the quick response.

**Comment by brainix** [comment]
Hi, @mlasevich.

I just released Pottery 1.3.6. Please upgrade to get this bug fix.

Thanks again for the feedback,
Raj
